### PR TITLE
docs: fix minor typos in various documentations

### PR DESCRIPTION
**Pull request by marcwrobel**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: docs/BAZEL.md [hunk 1]
```diff
@@ -239,7 +239,7 @@ yarn bazel analyze-profile filename_name.profile --html --html_details --html_hi
 
 This will generate a `filename_name.profile.html` file that you can open in your browser.
 
-On the upper right corner that is a small table of contents with links to three areas: Tasks, Legend and Statistics.
+In the upper right corner that is a small table of contents with links to three areas: Tasks, Legend and Statistics.
```
Comment (bob-watson) [inline docs/BAZEL.md:242]: Our style guide prefers the use of the Oxford comma in lists for clarity.
Suggest: `Tasks, Legend, and Statistics.` at the end of the sentence.

File: docs/DEVELOPER.md [hunk 2]
```diff
@@ -146,7 +146,7 @@ $ yarn lint
 ## Publishing Snapshot Builds
 
 When a build of any branch on the upstream fork angular/angular is green on CircleCI, it
-automatically publishes build artifacts to repositories in the Angular org, eg. the `@angular/core`
+automatically publishes build artifacts to repositories in the Angular org, e.g. the `@angular/core`
```
Comment (bob-watson) [inline docs/DEVELOPER.md:149]: `e.g.` should be `for example`

File: docs/PUBLIC_API.md [hunk 3]
```diff
@@ -15,7 +15,7 @@ Our semver, timed-release cycle and deprecation policy currently applies to thes
 - `@angular/service-worker`
 
 
-One intentional omission from this list is `@angular/compiler`, which is currently considered a low level api and is subject to internal changes. These changes will not affect any applications or libraries using the higher-level apis (the command line interface or JIT compilation via `@angular/platform-browser-dynamic`). Only very specific use-cases require direct access to the compiler API (mostly tooling integration for IDEs, linters, etc). If you are working on this kind of integration, please reach out to us first.
+One intentional omission from this list is `@angular/compiler`, which is currently considered a low level api and is subject to internal changes. These changes will not affect any applications or libraries using the higher-level apis (the command line interface or JIT compilation via `@angular/platform-browser-dynamic`). Only very specific use-cases require direct access to the compiler API (mostly tooling integration for IDEs, linters, etc.). If you are working on this kind of integration, please reach out to us first.
```
Comment (bob-watson) [inline docs/PUBLIC_API.md:18]: `api` should be capitalized throughout. If you need to refer to more than one, you can use: `APIs`.

**Review by bob-watson — CHANGES_REQUESTED** [review]
Thanks for the edits!
This looks good with only a few minor changes to make.

File: docs/BAZEL.md (hunk 1 above)
Comment (marcwrobel) [reply docs/BAZEL.md]: Done.

File: docs/DEVELOPER.md (hunk 2 above)
Comment (marcwrobel) [reply docs/DEVELOPER.md]: Done.

File: docs/PUBLIC_API.md (hunk 3 above)
Comment (marcwrobel) [reply docs/PUBLIC_API.md]: Done.

**Review by devversion — APPROVED** [review]

File: docs/DEVELOPER.md [hunk 4]
```diff
@@ -146,8 +146,8 @@ $ yarn lint
 ## Publishing Snapshot Builds
 
 When a build of any branch on the upstream fork angular/angular is green on CircleCI, it
-automatically publishes build artifacts to repositories in the Angular org, eg. the `@angular/core`
-package is published to https://github.com/angular/core-builds.
+automatically publishes build artifacts to repositories in the Angular org, for example the
```
Comment (bob-watson) [inline docs/DEVELOPER.md:149]: It just needs a comma after *for example*, then it LGTM.

**Review by bob-watson — CHANGES_REQUESTED** [review]
one more comma and it LGTM

File: docs/DEVELOPER.md (hunk 4 above)
Comment (marcwrobel) [reply docs/DEVELOPER.md]: I also split the sentence in two : `in the Angular org. For example, the`. I think it's easier to read that way. Do you think it is OK ?

**Review by bob-watson — APPROVED** [review]
LGTM!

**Review by AndrewKushnir — APPROVED** [review]

**Comment by AndrewKushnir** [comment]
@marcwrobel it looks like this PR has a conflict after merging some other PRs. Could you please rebase and resolve conflicts when you get a chance? Thank you.

**Comment by marcwrobel** [comment]
@AndrewKushnir, conflicts resolved.

**Comment by AndrewKushnir** [comment]
@marcwrobel thanks for resolving conflicts! I'm adding this PR to the merge queue.

**Comment by AndrewKushnir** [comment]
This PR was merged into the repository by commit 25c4e4cd5d8314f393837faf3f0c27d1496eb38a.

**Pull request closed without merging**